Problem with template argument

Excerpt of make output:

     ../../../SRC/.gcc-extra-ppl/src/Grid_Generator_System.defs.hh:272:28: error: template argument 3 is invalid
        const Grid_Generator&>,
                             ^
    
    Makefile:1159: recipe for target 'Scalar_Products.lo' failed

--- a/Makefile
+++ b/Makefile
@@ -246,6 +246,7 @@ all-src-gcc-extras = $(gcc-extras:%=SRC/.gcc-extra-%)
 $(all-src-gcc-extras): SRC/.gcc-extra-%: | SRC
 	rm -rf SRC/.gcc-extra-$*
 	cd SRC && tar xpf $(SRCDIR)/$(gcc_extra_dir)/$*-$(gcc_extra_version).tar.*
+	cd SRC && test -f $(SRCDIR)/$(gcc_extra_dir)/$*-$(gcc_extra_version).patch && cd $*-$(gcc_extra_version) && patch -p1 < ../$(SRCDIR)/$(gcc_extra_dir)/$*-$(gcc_extra_version).patch || true
 	mv SRC/$*-$(gcc_extra_version) SRC/.gcc-extra-$*
 
 # All Macs need Core2 assembly and --enable-fat is broken with stock MacOS gcc.

